### PR TITLE
Fix ShowClaimPage cannot render properly

### PR DESCRIPTION
**Pull request by ender503**
Because the 'ShowClaimPage' cannot render properly, we fix this problem
by the following steps: 
 - Fix claim transaction hash not pass correctly
 - Change ShowClaimPage into class-based component
 - Fix ShowClaimPage not render properly
 - Remove showClaims action

### CODE REVIEW

File: pages/claims/info.js [hunk 1]
```diff
@@ -49,4 +48,4 @@ const mapStateToProps = (state) => {
   return { user, claims };
 };
 
-export default withRedux(configureStore, mapStateToProps, mapDispatchToProps)(showClaimPage);
+export default withRedux(configureStore, mapStateToProps, mapDispatchToProps)(ShowClaimPage);
```
Comment (jserv) [inline pages/claims/info.js]: So, we never use camelCase for all commands, right?

File: constants/index.js [hunk 2]
```diff
@@ -24,10 +24,6 @@ export const CREATE_CLAIM_REQUEST = 'CREATE_CLAIM_REQUEST';
 export const CREATE_CLAIM_FAILURE = 'CREATE_CLAIM_FAILURE';
 export const CREATE_CLAIM_SUCCESS = 'CREATE_CLAIM_SUCCESS';
 
-export const SHOW_CLAIM_REQUEST = 'SHOW_CLAIM_REQUEST';
```
Comment (jserv) [inline constants/index.js]: Why did you remove these lines?

File: pages/claims/info.js [hunk 3]
```diff
@@ -1,52 +1,60 @@
...
-      <a>Description: {infoArray[4]}</a>
-    </Layout>
-  );
-};
-
-showClaimPage.propTypes = {
-  claims: PropTypes.array,
+import Layout from '../../layouts/material/Main';
+
+import { fetchClaimInfo } from '../../utils/tangleidAPI';
```
Comment (jserv) [inline pages/claims/info.js]: It might be annoying to specify `tangleidAPI`. Can it be feasible to get reduced into `API` or something alike?

File: pages/claims/info.js (hunk 1 above)
Comment (ender503) [reply pages/claims/info.js]: Yes, it is recommended to use lower camel case when naming variables and functions, and use upper camel case(pascal case) when naming classes.

reference: [Naming Conventions](https://github.com/airbnb/javascript#naming-conventions)

File: constants/index.js (hunk 2 above)
Comment (ender503) [reply constants/index.js]: Since we no longer to use the `showClaims` action, so we remove these related constants.

Redux = Action + Reducer
Redux holds the state in the one place.
When action dispatched, the reducer will call and change the Redux state.

Since the claim data only used in the `ShowClaimPage` page, the claim data is not necessary to behold in the Redux state.

The user profile and session token will suitable for Redux state when user login to our website.

Comment (jserv) [reply constants/index.js]: Indeed, mention in Git commit messages as well.

File: pages/claims/info.js (hunk 3 above)
Comment (ender503) [reply pages/claims/info.js]: I think this is necessary to specify `tangleidAPI` if we use more than one different APIs in our application.
Moreover, I'm planning to move the `tangleidAPI` to an NPM package, then everyone can use TangleID API. 😃

Comment (jserv) [reply pages/claims/info.js]: TangleID is conceptually an interface to bridge IOTA transactions and identity along with advanced features such MAM and extension-specific filters. The APIs provided by TangleID should be minimized as possible as we can, and the use cases like LASS or Taipei City Card would be no doubly built as extensions to TangleID. The more APIs exposed, the harder we can maintain in elegant ways.

Comment (ender503) [reply pages/claims/info.js]: Maybe we could take `iota.lib.js` as a [reference](https://github.com/iotaledger/iota.lib.js/#getting-started). 

Change the API as the following example. 🤔 
```
// Setup the TangleID instance
import TangleID from 'TangleID';
var tangleid = new TangleID({
    'provider': SWARM_NODE_ADDRESS
});

// Call TangleID API
tangleid.api.fetchUserList();

// Call extension APIs
tangleid.extenstion.method();
```

File: constants/index.js (hunk 2 above)
Comment (ender503) [reply constants/index.js]: I mention these into the commit message. 👍

File: pages/claims/info.js (hunk 3 above)
Comment (ender503) [reply pages/claims/info.js]: I'll open a new issue to describe this plan.

Comment (jserv) [reply pages/claims/info.js]: Let's ensure the consistency of API scheme first and then merge this pull request.

Comment (ender503) [reply pages/claims/info.js]: Roger that. 👍

Comment (jserv) [reply pages/claims/info.js]: This pull request depends on #54

**Comment by ender503** [comment]
Rebasing is required. 🐒

**Comment by ender503** [comment]
Branch is rebased.

**Comment by ender503** [comment]
Grammer mistake has been fixed. 👍
